### PR TITLE
feat(servicecatalog): add L2 constructs

### PR DESCRIPTION
**Pull request by JordanDeBeer**
This PR adds support for some AWS ServiceCatalog L2 constructs.

One piece of this PR that I think is particularly interesting and warrants a solid review is the `CDKProduct` resource. Particularly because it allows the user to pass in a `core.Stack` struct and the CDKProduct handles synthesizing it so it can use the generated template in an asset.

I don't fully understand how the `cdk-assets` package works enough to support assets, but I am willing to work this part further with some guidance.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-servicecatalog/lib/constraints.ts [hunk 1]
```diff
@@ -0,0 +1,329 @@
...
+}
+/**
+ * Launch Template Constraint properties
+ */
+export interface LaunchTemplateConstraintProps extends BaseConstraintProps {
+  // TODO: make this it's own type instead of a string
+  /**
+   * Launch Template Constraint Rules
+   */
+  readonly rules: string;
```
Comment (JordanDeBeer) [inline packages/@aws-cdk/aws-servicecatalog/lib/constraints.ts:104]: I wasn't able to find much information on how this works. The CfnResource `CfnLaunchTemplateConstraint` expects a string here.

File: packages/@aws-cdk/aws-servicecatalog/lib/constraints.ts [hunk 2]
```diff
@@ -0,0 +1,329 @@
...
+    this.stackInstanceControl = this.allowStackInstanceControl ? 'ALLOWED' : 'NOT_ALLOWED';
+    if (props.product && props.portfolio) {
+      this.apply(props.product, props.portfolio);
+    }
+  }
+  public apply(product: IProduct, portfolio: IPortfolio) {
+    new servicecatalog.CfnStackSetConstraint(this, 'stackSetConstraint', {
+      acceptLanguage: this.acceptLanguage,
+      // As of 2020-10-23 the CloudFormation Resource Specification says the "Description" parameter of this resource is required.
+      // This is likely a bug and these lines can be swapped once this issue is resolved.
```
Comment (JordanDeBeer) [inline packages/@aws-cdk/aws-servicecatalog/lib/constraints.ts:317]: This line is definitely worth paying attention to.

**Review by MrArnoldPalmer — CHANGES_REQUESTED** [review]
Just some preliminary feedback, specifics to come as I dig in.

Can you write a section in the module's README with some example usage of the new constructs. Specifically docs about `CdkProduct` vs `Product` I think would be helpful. They definitely would be for me 😄 .

We need to update the `maturity` and `stability` fields in the package.json to reflect that the `L2s` are experimental.

Very excited for this, TY!!

**Comment by JordanDeBeer** [comment]
Not sure why the build is failing due to missing "Experimental" banner.

**Pull request closed without merging**